Sources/Examples/ReflectionService/Documentation.docc/ReflectionServiceTutorial.md

```diff
@@ -0,0 +1,177 @@
+
+
+## Overview
+
+The Reflection service provides information about the public RPCs served by a Server. 
+It is specific to services defined using the Protocol Buffers IDL.
+By calling the Reflection service, clients can construct and send requests to services
+without needing to generate code and types for them. 
+
+You can also use CLI clients such as [gRPCurl][grpcurl-setup] and the [gRPC command line tool][grpc-cli] to: 
+- list services,
+- describe services and their methods,
+- describe symbols,
+- describe extensions,
+- construct and invoke RPCs.
+
+gRPC Swift supports both [v1][v1] and [v1alpha][v1alpha] of the  reflection service.
+
+## Adding the Reflection service to a server
+You can use the Reflection Service by adding it in the providers when constructing your server.
+
+To initialise the Reflection service we will use 
+``Sources/GRPCReflectionService/ReflectionService/init(serializedReflectionDataFilePaths:version:)``.
+It receives the paths to the files containing the reflection data of the proto files 
+describing the services of the sqerver that we want to be discoverable through reflection
+and the version of the reflection service.
+
+
+### Generating the reflection data
+The server from this example uses the `GreeterProvider` and the `EchoProvider`,
+besides the `ReflectionService`.
+  The associated proto files are located at 
+ `Sources/Examples/HelloWorld/Model/helloworld.proto`, and 
+ `Sources/Examples/Echo/Model/echo.proto` respectively.
+
+ In order to generate the reflection data for the
+ `helloworld.proto`, you can run the following command:
+
+```sh
+$ protoc Sources/Examples/HelloWorld/Model/helloworld.proto \
+    --proto_path=Sources/Examples/HelloWorld/Model \
+    --plugin=./.build/debug/protoc-gen-grpc-swift \
+    --grpc-swift_opt=Client=false,Server=false,ReflectionData=true \
+    --grpc-swift_out=Sources/Examples/ReflectionService/Generated
+```
+
+Let's break the command down:
+- The first argument passed to `protoc` is the path 
+to the `.proto` file to generate reflection data
+for: [`Sources/Examples/HelloWorld/Model/helloworld.proto`][helloworld-proto].
+- The `proto_path` flag is the path to search for imports: 
+`Sources/Examples/HelloWorld/Model`.
+- The plugin we are using is `./.build/debug/protoc-gen-grpc-swift`.
+- The `grpc-swift_opt` options are passed to the `protoc-gen-grpc-swift` plugin. 
+To generate only the reflection data set `Client=false,Server=false,ReflectionData=true`.
+- The `grpc-swift_out` flag is used to set the path of the directory
+where the generated file will be located: `Sources/Examples/ReflectionService/Generated`.
+
+The commands for generating the reflection data for the `Echo` and `Reflection` services are similar.
+
+Depending on the version of [gRPCurl][grpcurl] you are using you might need to use the v1alpha proto instead.
+Beginning with [gRPCurl v1.8.8][grpcurl-v188] it uses the [v1][v1] reflection. Earlier versions use [v1alpha][v1alpha]
+reflection.
```

Comment:
I think this part should come in the next section, after where you say "in our case `.v1`".